apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -124,4 +127,57 @@ defmodule Snitch.Domain.Order.Transitions do
   end
 
   def persist_shipment(%Context{valid?: false} = context), do: context
+
+  defp process_package(package, shipping_method_id) do
+    shipping_method =
+      Enum.find(package.shipping_methods, fn %{id: id} ->
+        id == shipping_method_id
+      end)
+
+    package_total =
+      Enum.reduce(
+        [shipping_method.cost],
+        &Money.add!/2
+      )
+
+    Package.update(package, %{
+      cost: shipping_method.cost,
+      total: package_total,
+      shipping_method: shipping_method
+    })
+  end
+
+  @doc """
+  Persists shipping_method_id to packages
+
+  Calculate pacakge total cost Sum(shipping_cost, adjustment_total,promo_total, shipping_cost)
+
+  Update package cost total in DB
+  """
+
+  @spec save_packages_methods(Context.t()) :: Context.t()
+  def save_packages_methods(
+        %Context{valid?: true, struct: %Order{id: order_id} = order} = context
+      ) do
+    %{state: %{selected_shipping_methods: selected_shipping_methods}, multi: multi} = context
+
+    packages = Repo.all(from(p in PackageSchema, where: p.order_id == ^order_id))
+
+    function = fn _ ->
+      selected_shipping_methods
+      |> Stream.map(fn %{package_id: package_id, shipping_method_id: shipping_method_id} ->
+        package = Enum.find(packages, fn package -> package.id == package_id end)
```

Comment:
I wouldn't mind doing `package = Package.get(package_id)` here (and not fetch them together), but it's fine.
More importantly, I'd either,
* like you to do some validations here (or somewhere):
  1. ensure that a `shipping_method` is chosen for all `packages` of this `order`.
  2. ensure that `package_ids` are unique (in the `selected_shipping_methods` map).
  3. what else is needed here @pkrawat1, @ashish173, @arjun289, @SagarKarwande ?
* like you to document assumptions about the "correctness" of this `map`, so that the validation can be written later/elsewhere.

Reply:
@oyeb most important is the first validation that you pointed out. we are good with these for now.